.github/workflows/e2e_test_reusable.yml

```diff
@@ -0,0 +1,92 @@
+on:
+  workflow_call:
+
+jobs:
+  prepare-matrix:
+    name: Prepare Matrix
+    runs-on: ubuntu-latest
+    # Define the outputs that this job will produce
+    outputs:
+        files: ${{ steps.set-files.outputs.FILES }}
+    steps:
+      # Checkout the code
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 #v4.1.1
+
+      # Set the files to run
+      - id: set-files
+        run: |
+          FILE_LIST=$(ls frontend/cypress/e2e/*.cy.js | jq -R -s -c 'split("\n")[:-1]')
+          echo "FILES=$FILE_LIST" >> $GITHUB_OUTPUT
+
+      # Print the output 'files' for testing
+      - name: Test Output
+        run: |
+          echo "${{ steps.set-files.outputs.FILES }}"
+
+  e2e:
+    name: End-to-End Testing
+    runs-on: ubuntu-latest
+    needs: prepare-matrix # Only run this job if the prepare-matrix job is successful
+
+    # Define the strategy for running the e2e job
+    strategy:
+      fail-fast: false # Continue running tests even if some fail
+      matrix:
+        file: ${{fromJson(needs.prepare-matrix.outputs.files)}}
+
+    steps:
+      # Checkout the code
+      - name: Checkout
+        uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4
+
+      # Install and cache NPM dependencies
+      - uses: ./.github/actions/setup-javascript
+
+      # Stand up the system stack, to have something to poke
+      - name: Launch Stack
+        uses: ./.github/actions/run-full-stack
+        env:
+          JWT_PRIVATE_KEY: ${{ secrets.JWT_PRIVATE_KEY }}
+
+      # Install Cypress globally
+      - name: Install Cypress globally
+        run: bun install -g cypress
+
+      # Extract test filename
+      - name: Get Filename
+        id: get-filename
+        run: |
+          TEST_FILE=$(basename "${{ matrix.file }}" | sed 's/\.cy\.js//')
+          echo "TEST_FILE=$TEST_FILE" >> $GITHUB_OUTPUT
+
+      # Run the Cypress E2E Tests
+      - name: E2E Test
+        working-directory: frontend
+        run: |
+          bun run cypress install && bun run test:e2e --spec $(echo "${{ matrix.file }}" | sed 's/frontend\///')
```

Comment:
The `sed` command changes the file path by taking out "frontend/" from the file name